src/api/db/models/Image.js

```diff
@@ -482,6 +465,85 @@ export class ImageModel {
     }
   }
 
+  /**
+   * This endpoint is used only by the ML Handler and allows labels to be Upserted
+   * onto the Project label list when necessary. Users cannot use this endpoint
+   *
+   * @param {object} input
+   * @param {object} context
+   */
+  static async createInternalLabels(input, context) {
+    console.log('ImageModel.createInternalLabels - input: ', JSON.stringify(input));
+    const operation = async ({ label }) => {
+      return await retry(async () => {
+        console.log('ImageModel.createInternalLabels - creating label: ', JSON.stringify(label));
+
+        label.type = 'ml';
+
+        // find image, create label record
+        const image = await ImageModel.queryById(label.imageId, context);
+        if (isLabelDupe(image, label)) throw new DuplicateLabelError();
+
+        const project = await ProjectModel.queryById(image.projectId);
+        const labelRecord = createLabelRecord(label, label.mlModel);
+
+        // Check if Label Exists on Project and if not, add it
+        if (!project.labels.some((l) => { return l._id === labelRecord.labelId; })) {
```

Comment:
Any time I'm doing `_id` equality checks w/ MongoDB _ids I always coerce them to strings first to make sure they're the same type. There's a helper function for this in utils.js, `idMatch()`, if you want to use it.